### PR TITLE
Strip thumbnail assets from payload before discovery

### PR DESCRIPTION
**Pull request by ividito**
Mutate payload before discovery to ignore thumbnail assets meant for the collection level.

### CODE REVIEW

File: dags/veda_data_pipeline/veda_dataset_pipeline.py [hunk 1]
```diff
@@ -61,6 +61,18 @@ def build_stac_task(payload):
...
+def mutate_payload(**kwargs):
+    ti = kwargs.get("ti")
+    payload = ti.dag_run.conf
+    if assets := payload.get("assets"):
+        # is first key thumbnail
+        if "thumbnail" in assets.keys():
+            assets.pop("thumbnail")
+        if not assets:
+            payload.pop("assets")
+    return payload
```
Comment (anayeaye) [inline dags/veda_data_pipeline/veda_dataset_pipeline.py]:
```suggestion
def mutate_payload(**kwargs):
    ti = kwargs.get("ti")
    payload = ti.dag_run.conf
    if assets := payload.get("assets"):
        # remove thumbnail asset if provided in collection config
        if "thumbnail" in assets.keys():
            assets.pop("thumbnail")
        # if thumbnail was only asset, delete assets
        if not assets:
            payload.pop("assets")
        # finally put the mutated assets back in the payload
        else:
            payload["assets"] = assets
    return payload
```

File: dags/veda_data_pipeline/veda_dataset_pipeline.py [hunk 2]
```diff
@@ -98,4 +115,5 @@ def build_stac_task(payload):
     submit_stac = submit_to_stac_ingestor_task.expand(built_stac=build_stac)
 
     collection_grp.set_upstream(start)
+    mutate_payload_task.set_upstream(start)
```
Comment (anayeaye) [inline dags/veda_data_pipeline/veda_dataset_pipeline.py:118]: should `collection_grp` aslo be upstream of mutate_payload_task so we don't strip assets before the collection is created?

Comment (ividito) [reply dags/veda_data_pipeline/veda_dataset_pipeline.py:118]: Nope - it's not an in-place mutation, the output gets passed to the discover step independantly.

Comment (anayeaye) [reply dags/veda_data_pipeline/veda_dataset_pipeline.py:118]: Resolving--the discover task, which includes the `alt_payload=mutate_payload_task()` parameter, depends on `collection_grp` already

**Review by anayeaye — APPROVED** [review]

**Pull request merged**